File: app/src/main/java/com/example/testlunu/di/modules/MvpModule.java
package com.example.testlunu.di.modules;

import android.content.Context;


import com.example.testlunu.net.RetrofitProvider;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class MvpModule {
    /*modules injectors*/
    private final Context context;
    public MvpModule(Context context){
        this.context = context;
    }

    @Provides
    @Singleton
    Context provideContext(){
        return context;
    }

    @Provides
    @Singleton
    RetrofitProvider provideRetrofit(){
        return new RetrofitProvider();
    }

//    @Provides
//    @Singleton
//    MainContract.Presenter provideMainNewsContractPresenter(Context context){
//        return new MainNewsPresenter( context);
//    }
//
//    @Provides
//    @Singleton
//    MainContract.Model provideMainNewsContractModel(Context context){
//        return new MainNewsModel(context);
//    }

}

File: app/src/main/java/com/example/testlunu/di/modules/MvpModule.kt
package com.example.testlunu.di.modules

import android.content.Context
import com.example.testlunu.main.MainContract
import com.example.testlunu.main.MainModel
import com.example.testlunu.main.MainPresenter
import com.example.testlunu.main.MainVActivity
import com.example.testlunu.net.RetrofitProvider
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class MvpModule(private val context: Context) {

    @Provides
    @Singleton
    fun provideContext(): Context = context

    @Provides
    @Singleton
    fun provideRetrofit(): RetrofitProvider = RetrofitProvider()

    @Provides
    @Singleton
    fun provideMainContractPresenter (context:Context): MainContract.Presenter = MainPresenter(context)


    @Provides
    @Singleton
    fun provideMainContractModel (context:Context): MainContract.Model = MainModel(context)


}

